project/src/main/webapp/script.js

```diff
@@ -38,6 +44,9 @@ function fetchFromQuery() {
   displayResultsPage();
   const userCoords = JSON.parse(localStorage.getItem('userLocation'));
   const map = createMap({lat: userCoords.lat, lng: userCoords.lng});
+  if (googleUser) {
+    registerUserByToken();
+  }
```

Comment:
Why does the registration phase happens when fetching from the server? 
I think it should happen directly after the signing is successful from Google. WDYT?

Reply:
I don't really mind, done

Reply:
I think it has implications on multiple things in the flow, such as UI responsiveness and latency for the user.